scripts/scil_compute_ihMT_maps.py

```diff
@@ -0,0 +1,320 @@
+    Aapp = ((2*acq_parameters[0][0]/(acq_parameters[0][1]**2))-(2*acq_parameters[1][0]/(acq_parameters[1][1]**2)))/(((2*acq_parameters[0][0])/(acq_parameters[0][1]*cPD1))-((2*acq_parameters[1][0])/(acq_parameters[1][1]*cT1)))
+    T1app = ((cPD1/acq_parameters[0][1])-(cT1/acq_parameters[1][1]))/((cT1*acq_parameters[1][1])/(2*acq_parameters[1][0])-(cPD1*acq_parameters[0][1])/(2*acq_parameters[0][0]))
+    MTsat = 100*(((Aapp*acq_parameters[0][1]*acq_parameters[0][0]/T1app)/cPD2)-(acq_parameters[0][0]/T1app)-(acq_parameters[0][1]**2)/2)
+
+    return MTR, MTsat
+
+
+def threshold_ihMT_maps(computed_map, contrast_maps, in_mask, lower_threshold,
+                        upper_threshold, idx_contrast_list):
+    """
+    Remove NaN, apply different threshold based on
+       - maximum and minimum threshold value
+       - combination of specific contrasts maps
+    Multiple data by probability T1 mask (GM+WM+CSF)
+
+    Parameters
+    ----------
+    computed_map        3D-Array data.
+                        Myelin map (ihMT or non-ihMT computed maps)
+    contrast_maps       List of 3D-Array. File must containing the
+                        5 contrasts maps.
+    in_mask             Path to probability T1 map
+    lower_threshold     Value for low thresold <int>
+    upper_thresold      Value for up thresold <int>
+    idx_contrast_list   List of indice contrasts maps in order corresponding to
+                        contrast_maps input ex.: [1, 2, 3]
+                        Altnp = 0; Atlpn = 1; Reference = 2; Negative = 3;
+                        Positive = 4; T1weighted = 5
+    Returns
+    ----------
+    Thresholded matrice in 3D-array.
+    """
+    # Remove NaN and apply thresold based on lower and upper value
+    computed_map[np.isnan(computed_map)] = 0
+    computed_map[np.isinf(computed_map)] = 0
+    computed_map[computed_map < lower_threshold] = 0
+    computed_map[computed_map > upper_threshold] = 0
+
+    # Load and apply probability masks on maps
+    mask_image = nib.load(in_mask)
+    mask_data = mask_image.get_fdata(dtype=np.float32)
+    computed_map[np.where(mask_data == 0)] = 0
+
+    # Apply threshold based on combination of specific contrasts maps
+    for idx in idx_contrast_list:
+        computed_map[contrast_maps[idx] == 0] = 0
+
+    return computed_map
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_output_dirs_exist_and_empty(parser, args, os.path.join(args.out_dir, 'Contrats_MT_maps'),
+                                       os.path.join(args.out_dir, 'ihMT_native_maps'),
+                                       create_dir=True)
+
+    # Select contrasts files from input folder
+    maps = []
+    acquisitions = ['altnp', 'altpn', 'mtoff', 'neg', 'pos', 'T1w']
+    for acquisition in acquisitions:
+        maps.append(sorted(glob.glob(os.path.join(args.in_folder,
+                                                  '*' + acquisition + '*gz'))))
+
```

Comment:
You should check if you have all your images and if you get only one file per glob ? Why do you sort ? Maybe make sure all the files have the same name but the acquisition name ? Don't do everything here I try to be more robust and strict

Reply:
Same for json files you should not think that they have to be there.
You can check you have 6 nii.gz and 6 json files. Again these are suggestion to make it more robust, you can find your own way.

Reply:
It's 6 nii.gz per echo you may find a way to get the number of echoes and check after you've got everything you need.